### PR TITLE
rules: skip line-length rule for URLs and quoted lines

### PR DESCRIPTION
**Pull request by addaleax**
Fixes: https://github.com/nodejs/core-validate-commit/issues/24

### CODE REVIEW

File: lib/rules/line-length.js [hunk 1]
```diff
@@ -31,6 +31,12 @@ module.exports = {
     var failed = false
     for (let i = 0; i < parsed.body.length; i++) {
       const line = parsed.body[i]
+      // Skip quoted lines, e.g. for riginal commit messages of V8 backports.
```
Comment (richardlau) [inline lib/rules/line-length.js]: `riginal`->`original`

Comment (addaleax) [reply lib/rules/line-length.js]: @richardlau thanks, done!

File: lib/rules/line-length.js [hunk 2]
```diff
@@ -31,6 +31,12 @@ module.exports = {
     var failed = false
     for (let i = 0; i < parsed.body.length; i++) {
       const line = parsed.body[i]
+      // Skip quoted lines, e.g. for original commit messages of V8 backports.
+      if (line.startsWith('  '))
+        continue
+      // Skip lines with URLs.
+      if (line.match(/https?:\/\//))
```
Comment (thefourtheye) [inline lib/rules/line-length.js]: I think RegExp#test would be better here.

File: lib/rules/line-length.js [hunk 3]
```diff
@@ -31,6 +31,12 @@ module.exports = {
     var failed = false
     for (let i = 0; i < parsed.body.length; i++) {
       const line = parsed.body[i]
+      // Skip quoted lines, e.g. for original commit messages of V8 backports.
+      if (line.startsWith('  '))
```
Comment (thefourtheye) [inline lib/rules/line-length.js]: Shouldn't this have four space characters? The test actually tests with four space characters.

Comment (addaleax) [reply lib/rules/line-length.js]: I’m not sure … switched to 4 spaces for now, yes

File: lib/rules/line-length.js (hunk 2 above)
Comment (addaleax) [reply lib/rules/line-length.js]: @thefourtheye There’s 0 difference here, right? Anyway, done!

Comment (thefourtheye) [reply lib/rules/line-length.js]: You are correct, there will be no difference. I was going for idiomatic code 🙂

**Comment by lundibundi** [comment]
@thefourtheye, @joyeecheung, @addaleax I think this can be landed, isn't it?
(Also, Fixes/Closes https://github.com/nodejs/core-validate-commit/issues/24 shoudl probably be added)

**Comment by joyeecheung** [comment]
@lundibundi Yes, I'll push the button

**Pull request merged**